core/serialization/deserializer/montage-interpreter.js

```diff
@@ -208,12 +212,8 @@ var MontageContext = Montage.specialize({
                 result;
 
             if (typeof reviver.didReviveObjects === "function") {
-                result = reviver.didReviveObjects(this._objects, this);
-                if (Promise.is(result)) {
-                    return result.then(function() {
-                        return self._objects;
-                    });
-                }
+                reviver.didReviveObjects(this._objects, this);
```

Comment:
what if the interpreter isn't synchronous?

Reply:
In practice, `didReviveObjects` never actually did any async work but was returning a promise anyway.

Reply:
In practice, `didReviveObjects` never actually did any async work but was returning a promise anyway.